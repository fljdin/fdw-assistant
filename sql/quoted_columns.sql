-- withkeywords table is used to test reserved keywords
-- in "extract" method
CREATE TABLE withkeywords1 (
    id serial primary key,
    "limit" integer not null
);

INSERT INTO withkeywords1 (id, "limit") VALUES (1, 1);

CREATE TABLE withkeywords2 (
    id serial primary key,
    "limit" integer not null
);

INSERT INTO config (source, target, pkey, condition, batchsize, trunc) VALUES
    ('withkeywords1', 'withkeywords2', 'id', null, null, true);

SELECT target, invocation FROM plan('{withkeywords2}');

-- copy(1) should succeed and "limit" must be quoted
CALL copy(1);
