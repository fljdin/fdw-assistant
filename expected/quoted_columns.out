-- withkeywords table is used to test reserved keywords
-- in "extract" method
CREATE TABLE withkeywords1 (
    id serial primary key,
    "limit" integer not null
);
CREATE TABLE
INSERT INTO withkeywords1 (id, "limit") VALUES (1, 1);
INSERT 0 1
CREATE TABLE withkeywords2 (
    id serial primary key,
    "limit" integer not null
);
CREATE TABLE
INSERT INTO config (source, target, pkey, condition, batchsize, trunc) VALUES
    ('withkeywords1', 'withkeywords2', 'id', null, null, true);
INSERT 0 1
SELECT target, invocation FROM plan('{withkeywords2}');
    target     |       invocation        
---------------+-------------------------
 withkeywords2 | CALL assistant.copy(1);
(1 row)

-- copy(1) should succeed and "limit" must be quoted
CALL copy(1);
NOTICE:  Executing: TRUNCATE withkeywords2
NOTICE:  Executing: SELECT count(id) FROM withkeywords1 WHERE id > 0 
NOTICE:  Executing: INSERT INTO withkeywords2 SELECT id, "limit" FROM withkeywords1 WHERE id > 0  ORDER BY id 
CALL
